source/val/validate_image.cpp

```diff
@@ -830,6 +830,36 @@ spv_result_t ValidateSampledImage(ValidationState_t& _,
                << "' as an operand of <id> '"
                << _.getIdName(consumer_instr->id()) << "'.";
       }
+
+      if (consumer_opcode != SpvOpSampledImage &&
```

Comment:
Can you turn this into a function and write it as a switch statement?

Reply:
Done